### PR TITLE
Post "Read the source"

### PR DESCRIPTION
**Pull request by maelle**
cf #12

### CODE REVIEW

File: content/post/2019-05-16-read-the-source.Rmarkdown [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  - CRAN
+---
+
+Ever heard the phrase "Read the source, Luke"? It's a play on ["Use the force, Luke" from Star Wars](https://mygeekwisdom.com/2012/04/14/use-the-force-luke/), with no definite _source_ :wink:, [maybe a blog post by Jeff Atwood](https://blog.codinghorror.com/learn-to-read-the-source-luke/); and it underlines how important and useful it can be to read the source code of a tool instead of just its docs. 
+
+In this blog post, we shall explain why and how to read the source code of your R tools be they base R or packages, and how a R-hub service is part of the reason why this process has gotten easier.
+
+# Why read the source?
+
+In which cases would you want to read the actual code of a function or of a whole package? Here a few that come to mind:
```
Comment (gaborcsardi) [inline content/post/2019-05-16-read-the-source.Rmarkdown]: here _are_ a few

File: content/post/2019-05-16-read-the-source.Rmarkdown [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+date: '2019-05-12'
+slug: read-the-source
+tags:
+  - help
+  - CRAN
+---
+
+Ever heard the phrase "Read the source, Luke"? It's a play on ["Use the force, Luke" from Star Wars](https://mygeekwisdom.com/2012/04/14/use-the-force-luke/), with no definite _source_ :wink:, [maybe a blog post by Jeff Atwood](https://blog.codinghorror.com/learn-to-read-the-source-luke/); and it underlines how important and useful it can be to read the source code of a tool instead of just its docs. 
+
+In this blog post, we shall explain why and how to read the source code of your R tools be they base R or packages, and how a R-hub service is part of the reason why this process has gotten easier.
```
Comment (gaborcsardi) [inline content/post/2019-05-16-read-the-source.Rmarkdown]: comma after "tools"?

how _an_ R-hub service

File: content/post/2019-05-16-read-the-source.Rmarkdown [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+rhub::check_for_cran
+```
+
+The snippet above will open the source code of the function, check-cran.R. 
+```r
+lookup(body)
+lookup_browse()
+```
+
+This last snippet will first show the body of the `body()` function locally, and then open a browser window pointing at it in the GitHub mirror of R source.
```
Comment (gaborcsardi) [inline content/post/2019-05-16-read-the-source.Rmarkdown]: of _the_ R source.

File: content/post/2019-05-16-read-the-source.Rmarkdown [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+This last snippet will first show the body of the `body()` function locally, and then open a browser window pointing at it in the GitHub mirror of R source. 
+
+All in all, `lookup` is a handy package. :ok_hand: How does it work under the hood? We'll let you read its source, but part of the magic is supported by mirrors of R code hosted on GitHub:
+
+* [mirror of R source code](https://github.com/wch/r-source),
+
+* [mirror of CRAN packages](https://github.com/cran), provided by R-hub :sunglasses:,
+
+* [mirror of Bioconductor](https://github.com/Bioconductor/mirror).
```
Comment (gaborcsardi) [inline content/post/2019-05-16-read-the-source.Rmarkdown]: This is gone, unfortunately: https://github.com/Bioconductor-mirror does not exist any more.

File: content/post/2019-05-16-read-the-source.Rmarkdown [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+# How to search the source
+
+What if you don't know whose source code you'd like to read, i.e. you'd like to see how `sapply()` is used in the wild? That's another case where the code mirrors mentioned previously can help you! All these mirrors are based on GitHub repos, so to use them, you can
```
Comment (gaborcsardi) [inline content/post/2019-05-16-read-the-source.Rmarkdown]: Can we search for `vapply()`? `sapply()` should not be used in new code.

File: content/post/2019-05-16-read-the-source.Rmarkdown [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+* use [GitHub (advanced) search](https://help.github.com/en/articles/about-searching-on-github)
+
+* use [GitHub search via its V3 API](https://developer.github.com/v3/search/) and [sthe `gh` package](https://github.com/r-lib/gh), but beware of rate limits.
```
Comment (gaborcsardi) [inline content/post/2019-05-16-read-the-source.Rmarkdown]: sthe -> the

File: content/post/2019-05-16-read-the-source.Rmarkdown [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+```r
+lookup::lookup_usage("sapply")
+```
+
+will open https://github.com/search?l=r&q=%22sapply%22+user%3Acran+language%3AR&ref=searchresults&type=Code&utf8=%E2%9C%93, i.e. a GitHub search URL, restricted to R-hub's CRAN source code mirror.
+
+# Conclusion
+
+Reading the R source is useful, and the process of finding it got smoother thanks to the `lookup` package. A piece of the machinery behind it is R-hub's CRAN source code mirror, about which you can find out more in [its source](https://github.com/r-hub/cranatgh/) and in [our docs](https://docs.r-hub.io/#cranatgh). And once you're browsing our docs, stay a while and have a look at our other free services helping R package development!
```
Comment (gaborcsardi) [inline content/post/2019-05-16-read-the-source.Rmarkdown]: no need to link to cranatgh, that's all internal stuff.

Comment (maelle) [reply content/post/2019-05-16-read-the-source.Rmarkdown]: even in a post about the source? ok :wink:

File: content/post/2019-05-16-read-the-source.Rmarkdown (hunk 4 above)
Comment (maelle) [reply content/post/2019-05-16-read-the-source.Rmarkdown]: Ah! Good catch!

**Comment by maelle** [comment]
Thanks for the corrections! Ok for me to publish it now?

**Comment by gaborcsardi** [comment]
Sure!

**Pull request merged**